### DISCUSSION TITLE
Configure settings in app event hooks

### DISCUSSION THREAD

**Discussion by stuartin** [pocketbase/pocketbase#4470] [Q&A]
Firstly, thanks for all your hard work, I love Pocketbase! 🙏

I'm trying to using the application event hooks to configure some settings (name, SMTP, backups, storage etc.).
I've previously been doing this using migrations, however I'd like to move this to the application eventhooks.

I've tried using both the `onBeforeBootstrap` and `onAfterBootstrap` but cant seem to get either working.
From reading https://github.com/pocketbase/pocketbase/discussions/2254, I believe the correct hook will be the `onAfterBootstrap` and the relevant database connection and tables will have been created.

The below is what I've tried and the relevant error for each. This is in a `pb_hooks/bootstrap.pb.js`  file (I have some other hooks working without problems) on version `0.21.1`.

```js
onAfterBootstrap((e) => {
	// const settings = new SettingsUpsertForm(e.app);
	// settings.meta.appName = 'TEST';
	// settings.submit();
	// GoError: SQL logic error: no such table: _params (1) at reflect.methodValueCall (native)

	// const dao = e.app.dao();
	// const settings = dao.findSettings();
	// settings.meta.appName = 'TEST';
	// dao.saveSettings();
	// GoError: SQL logic error: no such table: _params (1); failed query: SELECT {{_params}}.* FROM `_params` WHERE `key`={:p0} LIMIT 1 at reflect.methodValueCall (native)
})

```

Any ideas?

**Comment by ganigeorgiev** [comment]
`onAfterBootstrap` runs **on every app startup** and it is not recommended for changing the default settings because it could overwrite any modification done to the settings from the Admin UI on restart.

For such cases prefer migrations as they run only once - https://pocketbase.io/docs/js-migrations/#initialize-default-application-settings.

**Reply by stuartin** [reply]
Thanks @ganigeorgiev for the quick reply.
This is actually what I'd prefer, in a production environment I'd rather enforce the configuration via code/config so any changes via clickop's (through the admin UI) are invalidated when we start pocketbase and configuration is stored and tracked in `git`.

Is it technically possible to do this in the `onAfterBootstrap` hook?

I was also playing around with trying to add a custom cli command as a workaround if its not possible to do this in the hook, but settings also dont seem to be preserved/saved, not sure if thats possible either (command runs, but when I login to the UI, the appName is not set to `TEST`)?

```js
$app.rootCmd.addCommand(
	new Command({
		use: 'settings',
		run: (cmd, args) => {
			const dao = $app.dao();
			const settings = dao.findSettings();

			settings.meta.appName = 'TEST';

			dao.saveSettings(settings);
		}
	})
);
```
